### PR TITLE
init commit

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
See Readme

### CODE REVIEW

**Comment by aaron-steinfeld** [comment]
Example in use - https://github.com/hypertrace/attribute-service/pull/172

**Review by avinashkolluru — DISMISSED** [review]

File: src/main/java/org/hypertrace/gradle/dependency/HypertraceDependencySettingsPlugin.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+  private DependencyPluginProjectExtension addProjectExtension(Project project) {
+    return project
+        .getExtensions()
+        .create(
+            DependencyPluginProjectExtension.EXTENSION_NAME,
+            DependencyPluginProjectExtension.class);
+  }
+
+  private void addVersionCatalog(
```
Comment (avinashkolluru) [inline src/main/java/org/hypertrace/gradle/dependency/HypertraceDependencySettingsPlugin.java:143]: Will help a great deal if you can add a comment with the snippet of the changes this is automatically making to the settings and build.gradle files.

Comment (avinashkolluru) [reply src/main/java/org/hypertrace/gradle/dependency/HypertraceDependencySettingsPlugin.java:143]: I'm assuming this is similar to adding something like this manually in the settings file
```
versionCatalogs {
        create("commonLibs") {

```

Comment (aaron-steinfeld) [reply src/main/java/org/hypertrace/gradle/dependency/HypertraceDependencySettingsPlugin.java:143]: Sure, let me go back and add some comments through here. (and yes it is, those specifics lines map to 124-125).

**Review by avinashkolluru — APPROVED** [review]

**Pull request merged**